blueflood-core/src/test/java/com/rackspacecloud/blueflood/eventemitter/EmitterTest.java

```diff
@@ -0,0 +1,310 @@
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.eventemitter;
+
+import com.rackspacecloud.blueflood.concurrent.ThreadPoolBuilder;
+import junit.framework.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.*;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public class EmitterTest {
+
+    final String testEventName = "test";
+    final String testEventName2 = "test2";
+
+    List<RollupEvent> store;
+    List<RollupEvent> store2;
+    Emitter<RollupEvent> emitter;
+    Emitter.Listener<RollupEvent> listener;
+    Emitter.Listener<RollupEvent> listener2;
+    RollupEvent event1;
+    RollupEvent event2;
+
+    @Before
+    public void setUp() {
+        store = Collections.synchronizedList(new ArrayList<RollupEvent>());
+        store2 = Collections.synchronizedList(new ArrayList<RollupEvent>());
+        emitter = new Emitter<RollupEvent>();
+        listener = new EventListener(store);
+        listener2 = new EventListener(store2);
+        event1 = new RollupEvent(null, null, "event1", "gran", 0);
+        event2 = new RollupEvent(null, null, "event2", "gran", 0);
+    }
+
+    @Test
+    public void testSubscription() {
+        //Test subscription
+        emitter.on(testEventName, listener);
+        Assert.assertTrue(emitter.listeners(testEventName).contains(listener));
+    }
+
+    @Test
+    public void testConcurrentEmission() throws InterruptedException, ExecutionException {
+        //Test subscription
+        emitter.on(testEventName, listener);
+        //Test concurrent emission
+        ThreadPoolExecutor executors = new ThreadPoolBuilder()
+                .withCorePoolSize(2)
+                .withMaxPoolSize(3)
+                .build();
+        final RollupEvent obj1 = new RollupEvent(null, null, "payload1", "gran", 0);
+        final RollupEvent obj2 = new RollupEvent(null, null, "payload2", "gran", 0);
+        final CountDownLatch startLatch = new CountDownLatch(1);
+        Future<Object> f1 = executors.submit(new Callable<Object>() {
+            @Override
+            public Object call() throws Exception {
+                startLatch.await();
+                emitter.emit(testEventName, obj1);
+                return null;
+            }
+        });
+        Future<Object> f2 = executors.submit(new Callable<Object>() {
+            @Override
+            public Object call() throws Exception {
+                startLatch.await();
+                emitter.emit(testEventName, obj2);
+                return null;
+            }
+        });
+        Thread.sleep(1000);
```

Comment:
Because the person who wrote that particular test either doesn't know how to test multithreaded code, or were constrained to test it the wrong way due to bad architecture. It looks like he's giving the threads enough time to reach the `startLatch.await()` calls.